packages/dev/core/src/Materials/meshDebugPluginMaterial.ts

```diff
@@ -0,0 +1,442 @@
+                { name: "m_uvScale", size: 1, type: "float" },
+                { name: "m_uvColor1", size: 3, type: "vec3" },
+                { name: "m_uvColor2", size: 3, type: "vec3" },
+                { name: "m_materialColor", size: 3, type: "vec3" },
+            ],
+            fragment: `
+                uniform vec3 m_shadedDiffuseColor;
+                uniform vec3 m_shadedSpecularColor;
+                uniform float m_shadedSpecularPower;
+
+                #if MODE == 0 || MODE == 1
+                    uniform float m_wireframeThickness;
+                    uniform vec3 m_wireframeColor1;
+                #endif
+                #if MODE == 1
+                    uniform float m_vertexRadius;
+                    uniform vec3 m_wireframeColor2;
+                #elif MODE == 2 || MODE == 3
+                    uniform float m_uvScale;
+                    uniform vec3 m_uvColor1;
+                    uniform vec3 m_uvColor2;
+                #elif MODE == 5
+                    uniform vec3 m_materialColor;
+                #endif
+            `,
+        };
+    }
+
+    /**
+     * Bind the uniform buffer
+     * @param uniformBuffer Uniform buffer
+     */
+    public bindForSubMesh(uniformBuffer: UniformBuffer): void {
+        if (this._isEnabled) {
+            uniformBuffer.updateVector3("m_shadedDiffuseColor", this._options.shadedDiffuseColor!);
+            uniformBuffer.updateVector3("m_shadedSpecularColor", this._options.shadedSpecularColor!);
+            uniformBuffer.updateFloat("m_shadedSpecularPower", this._options.shadedSpecularPower!);
+            uniformBuffer.updateFloat("m_wireframeThickness", this._options.wireframeThickness!);
+            uniformBuffer.updateVector3("m_wireframeColor1", this._options.wireframeColor1!);
+            uniformBuffer.updateVector3("m_wireframeColor2", this._options.wireframeColor2!);
+            uniformBuffer.updateFloat("m_vertexRadius", this._options.vertexRadius!);
+            uniformBuffer.updateFloat("m_uvScale", this._options.uvScale!);
+            uniformBuffer.updateVector3("m_uvColor1", this._options.uvColor1!);
+            uniformBuffer.updateVector3("m_uvColor2", this._options.uvColor2!);
+            uniformBuffer.updateVector3("m_materialColor", this._options.materialColor!);
+        }
+    }
+
+    /**
+     * Get shader code
+     * @param shaderType "vertex" or "fragment"
+     * @returns Shader code
+     */
+    public getCustomCode(shaderType: string): Nullable<{ [pointName: string]: string }> {
+        return shaderType === "vertex"
+            ? {
+                CUSTOM_VERTEX_DEFINITIONS: `
+                attribute float initialPass;
+
+                varying vec3 vBarycentric;
+                flat varying vec3 vVertexWorldPos;
+                flat varying float vPass;
+            `,
+                CUSTOM_VERTEX_MAIN_END: `
+                float vertexIndex = mod(float(gl_VertexID), 3.);

```

Comment:
The variables in the global scope should also have a prefix (but the variables local to a function or a `{ ... }` block of code don't have to).